dev/src/pages/group.tsx

```diff
@@ -1,11 +1,11 @@
 import { Component, For, createSignal } from "solid-js";
 import { TransitionGroup } from "solid-transition-group";
 
-function shuffle<T extends any[]>(array: T): T {
+function shuffle<T>(array: T[]): T[] {
```

Comment:
There is a slight difference between the two. [(example)](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwGcALZRRCEARgB4AVeEADwxFWAPilQE8BtAXQB8AClgwo3AFzxaASmm0AUItCRYCFOmx5CJMhQBMdEWIkKB8mQOWroceGDwEM8VMgC2AIxAwA+mOk3Lx9rW3UHJxdnGCxUAHN-GBhpaNi40PA7BEdUZ1cPbz9U+MTk+GFiuPgAH3zgmFkMtXscvKDC0t9K0pSMGPiBGrrC60VWlzgCZAgMX0o5+ABeXVJyKmF2n1LZRQB6XfhDw4A9AH4xyPhJ6dn5gyWV-XXusR39o5Pz8auQKZm53wAZgexFWFEoGwKWxeSTeBw+ZwuuQmvxuAIALCC9GsIZs-GIun00ts9vCjoikXlrv8DAtlqCngZIfUSe8EV9LtTZrT7vTsYYKkSSq9SR94IjvlzfLTgXywSAmXjCf0EiK2eSOcifn9ub5MXLGcyOgSYQ1ReygA)
This is why I usually stick to the first one.
But for a dome it doesn't really matter

Reply:
That's true - I'd say T[] is generally more desirable because you can directly access the array element type.

Reply:
alright, works for me